frontend/src/views/DonationForm.vue

```diff
@@ -0,0 +1,245 @@
+        </span>
+      </v-layout>
+    </v-form>
+  </v-container>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from "vue-property-decorator";
+import firebase from "firebase/app";
+import fb from "../firebaseApp";
+import VuePhoneNumberInput from "vue-phone-number-input";
+import "vue-phone-number-input/dist/vue-phone-number-input.css";
+
+@Component({
+  name: "DonationForm",
+  components: { VuePhoneNumberInput }
+})
+export default class DonationForm extends Vue {
+  date = new Date().toISOString().substr(0, 10);
+  currencies = ["INR", "USD", "EUR", "RUB", "AUD", "YEN"];
+  menu = false;
+  amountRules = [(v: string) => (v && !!v.match(/^\d*$/)) || "Numbers only"];
+  emailRules = [
+    (value: string) => {
+      if (value && value.length > 0) {
+        const pattern = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+        return pattern.test(value) || "Invalid e-mail";
+      }
+      return "Required email";
+    }
+  ];
+  rules = [(v: string) => (v && v.length > 0) || "Required"];
+  form = {
+    date: new Date().toISOString().substr(0, 10),
+    donor: {
+      phoneNumber: ""
+    },
+    sum: {
+      currency: "INR"
+    },
+    collectedBy: "",
+    comment: ""
+  };
+  phone = null;
+  validPhone = false;
+  phoneError = "";
+  isSubmitting = false;
+  submissionStatus = "";
+
+  getSubmissionData() {
+    return {
+      ...this.form,
+      donor: {
+        ...this.form.donor,
+        phoneNumber: this.phone
+      },
+      timestamp: firebase.database.ServerValue.TIMESTAMP,
+      token: this.$route.params.token
+    };
+  }
+
+  validatePhone() {
+    if (this.validPhone) {
+      this.phoneError = "";
+      return true;
+    } else {
+      this.phoneError = "Invalid phone";
+      return false;
+    }
+  }
+
+  get submissionMessage() {
+    let message = "";
+    if (this.submissionStatus === "success") {
+      message = "Submission successful";
+    } else if (this.submissionStatus === "error") {
+      message = "Submission error";
+    }
+    return message;
+  }
+
+  clearForm() {
+    if (this.submissionStatus === "success") {
+      (this.$refs.form as any).resetValidation();
+      this.form = {
+        ...this.form,
+        donor: {
+          phoneNumber: ""
+        },
+        sum: {
+          currency: "INR"
+        },
+        comment: ""
+      };
+    }
+  }
+
+  async submitForm() {
+    if (this.isSubmitting) return;
+    this.submissionStatus = "";
+    const data = this.getSubmissionData();
+    const isValidForm = (this.$refs.form as any).validate();
+    const isValidPhone = this.validatePhone();
+    if (isValidForm && isValidPhone) {
+      this.isSubmitting = true;
+      await fb
+        .database()
+        .ref("/donations/cash")
+        .push()
+        .set(data)
+        .catch(() => {
+          this.submissionStatus = "error";
```

Comment:
Can we have more details here? Would be useful to get the error message.

Reply:
Firebase does not give much detail at most it says "permission denied".